netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -110,6 +111,14 @@
   private final Random random = new Random();
   private WriteQueue clientWriteQueue;
   private Http2Ping ping;
+  private boolean pinging = false; 
+  
+  @VisibleForTesting
+  int dataSinceLastPing = 0;
```

Comment:
These should be exposed as getters, since outside code should not be able to mutate them.